crates/core/src/dht/dht_reducers.rs

```diff
@@ -186,21 +190,7 @@ pub fn reduce_remove_queued_holding_workflow(
     let action = action_wrapper.action();
     let pending = unwrap_to!(action => Action::RemoveQueuedHoldingWorkflow);
     let mut new_store = (*old_store).clone();
-    if let Some(PendingValidationWithTimeout { pending: front, .. }) =
-        new_store.queued_holding_workflows.front()
-    {
-        if front == pending {
-            let _ = new_store.queued_holding_workflows.pop_front();
-        } else {
-            // The first item in the queue could be a delayed one which will result
-            // in the holding thread seeing another item as the next one.
-            // The holding thread will still try to pop that next item, so we need
-            // this else case where we just remove an item from some position inside the queue:
-            new_store
-                .queued_holding_workflows
-                .retain(|PendingValidationWithTimeout { pending: item, .. }| item != pending);
-        }
-    } else {
+    if let None = new_store.remove_holding_workflow(pending) {
         error!("Got Action::PopNextHoldingWorkflow on an empty holding queue!");
     }
```

Comment:
I replaced this big chunk of code with a simpler method on DhtStore. It seemed unnecessary to have two separate cases for whether the item is at the front of the queue or not, either way we just want to remove it. Can someone make sure I did this replacement right though? @lucksus?

Reply:
Looks good!